_posts/2019-10-15-dependency-management.md

```diff
@@ -0,0 +1,177 @@
+
+The proposal revolves around 3 axes:
+
+* improving the documentation about how versions are reconciled in sbt, and what to do about conflicts,
+* have a usable strict conflict manager in sbt, and
+* develop tools relying on static analysis to detect runtime conflicts between libraries that would not have been detected otherwise.
+
+At the time we started this project, Ivy was still used by default to resolve libraries in sbt, but efforts were in progress to use coursier instead, so we focused on the latter as a vehicle for improvements.
+Since then, coursier support in sbt [became official](https://github.com/sbt/sbt/pull/4614), and ships by default in sbt 1.3.0.
+Besides having a simpler model than the one used before in sbt or in Ivy, coursier has the advantage of being usable from the command line, and can benefit other users of the coursier API.
+
+For the third point about static analysis, we found [the missinglink tool](https://github.com/spotify/missinglink) for Maven, originating from Spotify themselves.
+This tool can analyze the classpath of a Maven project, and detect ahead of time (at "link" time) whether binary incompatibilities can be encountered at run-time.
+The only issue was that it was not usable from sbt, rendering it useless for a large subset of Scala users.
+We worked on [an sbt plugin for missinglink](https://github.com/scalacenter/sbt-missinglink) to address this hole.
+
+## Version ordering and reconciliation
+
+While most people have a simple intuition about library resolution (a newer version of a library get picked over an older version), the details of how versions are compared and reconciled are quite complicated.
+We contributed [a dedicated documentation page](https://get-coursier.io/docs/other-version-handling) on the coursier website to help users understand how resolution works in the not-so-trivial cases.
+In order to improve the experience of upgrading from sbt 1.2.x and Ivy-based resolution, we [tweaked resolution in coursier 2.0.0-RC3-4](https://github.com/coursier/coursier/pull/1348) to be more in line with Ivy, the former sbt behavior, and the [semantic versioning specification](https://semver.org).
+The documentation page should be updated accordingly soon.
+
+On top of that, Eugene Yokota recently wrote [a blog post](http://eed3si9n.com/dependency-resolver-semantics) comparing how versions are compared and reconciled in both Ivy and coursier.
+
+## Strict conflict manager
+
+By default, both Ivy and coursier tolerate that the dependency graph contains two different versions of a library, in which case they pick a winner.
+Sometimes, we want more control, and do not want any conflict resolution to happen.
+Instead, we would like an error to be reported if there are two different versions of a library.
+A *strict* conflict manager does precisely that, and we added on to coursier.
+
```

Comment:
I would suggest inserting here:

> Instructions follow for enabling the strict conflict manager 1) from the Coursier command line, and 2) in an sbt build.

I would also suggest reversing the order so the sbt instructions come first, as I think sbt is the most common case.